src/core/thread/mle.cpp

```diff
@@ -520,6 +520,7 @@ ThreadError Mle::SetRloc16(uint16_t aRloc16)
     }
 
     mMac.SetShortAddress(aRloc16);
+    mNetif.GetNetworkDataLocal().SendServerDataNotification();
```

Comment:
Okay, I updated this PR with a different approach.  Now a call is made at the end of `MleRouter::HandleAdvertisement()`, which should handle the case of updating routes based on receiving an Advertisement message.  The RFD case should already be handled with the call in `Mle::HandleNetifStateChanged()`.  Does this work?